.github/workflows/build.yml

```
@@ -0,0 +1,112 @@
+name: Build
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  ubuntu-gcc:
+    name: Ubuntu 18.04 / gcc
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+    - name: Install build dependencies
+      run: |
+        sudo apt-get update && sudo apt-get install -y --no-install-recommends \
+          ccache \
+          libavformat-dev \
+          libchromaprint-dev \
+          libebur128-dev \
+          libfaad-dev \
+          libfftw3-dev \
+          libflac-dev \
+          libid3tag0-dev \
+          liblilv-dev \
+          libmad0-dev \
+          libmodplug-dev \
+          libmp3lame-dev \
+          libmp4v2-dev \
+          libopus-dev \
+          libopusfile-dev \
+          libportmidi-dev \
+          libprotobuf-dev \
+          libqt5opengl5-dev \
+          libqt5sql5-sqlite \
+          libqt5svg5-dev \
+          libqt5x11extras5-dev \
+          librubberband-dev \
+          libshout3-dev \
+          libsndfile1-dev \
+          libsoundtouch-dev \
+          libsqlite3-dev \
+          libtag1-dev \
+          libupower-glib-dev \
+          libusb-1.0-0-dev \
+          libwavpack-dev \
+          portaudio19-dev \
+          protobuf-compiler \
+          qt5-default \
+          qt5keychain-dev \
+          qtscript5-dev \
+    - name: Set up cmake
+      uses: jwlawson/actions-setup-cmake@v1.4
+      with:
+        cmake-version: '3.13.x'
+    - name: Create build directory
+      run: mkdir cmake_build
+    - name: Prepare ccache timestamp
+      id: ccache_cache_timestamp
+      shell: cmake -P {0}
+      run: |
+        string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+        message("::set-output name=timestamp::${current_date}")
+    - name: Set up ccache cache
+      uses: actions/cache@v1.1.0
+      with:
+        path: .cache/ccache
+        key: ${{ github.job }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
```

```suggestion
        key: ${{ github.job }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
```
Putting the timestamp in the cache key is redundant. GitHub Actions already compares caches based on modification time. The cache key should contain an identifying string for the job and the branch. See https://github.com/mixxxdj/mixxx/pull/3305/files#diff-ba2a15f6a0557b1f58421c9e2f89dedd770ebb0ec79eb3be817268246ece1959R36